.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
```

Comment:
@tdeekens I'm still unsure what to use exactly as trigger events.
I understand that `push` is triggered by each commit and `pull_requests` is triggered by commits pushed to an open PR.
So if a PR is open, the workflow is run twice.
If we only use `pull_request`, the workflow is run only when you open a PR, so there might be cases where the workflow is skipped.
The `push` event is still important on `master` though.

How would you set this up so that it "works as expected" and it does not run twice?

Reply:
@emmenko yeah I am also still wrapping my head around those events. Generally, I would not want CI to run when I push to a branch is does not have a respectively open PR. Not necessarily as I might just want to "stash" my work upstream. Moreover, it cuts into the allowance or minutes on GitHub Actions often without having too I think.

That said, I would use `pull_request` on the general testing, building and linting workflow. As that then doesn't run on master I would add another workflow (or configure the same with two events) to have `push` but only run on `commercetools/commercetools-docs-kit/master`. At least that's how I got it closest to what I felt was good.

Reply:
Yeah ok.

The way it's configured now it's as you mentioned: triggered on `pull_request` events and `push` on `master` only.
So it should be good like this and does it meet your expectations?

Reply:
Yeah, it matches what I'd expect. I wouldn't expect CI to run without a PR being opened. But when I open only on the last commit. That may vary from person to person but still feels like a decent default.